File: app/src/main/java/com/example/pms/data/models/Project.kt
package com.example.pms.data.models

import java.util.Date
import java.util.UUID

data class Project(
    val id: String = UUID.randomUUID().toString(),
    val name: String,
    val description: String,
    val startDate: Date?,
    val endDate: Date?,
    val status: ProjectStatus,
    val createdAt: Date = Date()
)

enum class ProjectStatus {
    NOT_STARTED,
    IN_PROGRESS,
    ON_HOLD,
    COMPLETED,
    CANCELLED
}

File: src/models/Project.ts
import uuid from 'react-native-uuid';

export enum ProjectStatus {
  NOT_STARTED = 'NOT_STARTED',
  IN_PROGRESS = 'IN_PROGRESS',
  ON_HOLD = 'ON_HOLD',
  COMPLETED = 'COMPLETED',
  CANCELLED = 'CANCELLED'
}

export interface Project {
  id: string;
  name: string;
  description: string;
  startDate: Date | null;
  endDate: Date | null;
  status: ProjectStatus;
  createdAt: Date;
}

export const createProject = (
  name: string, 
  description: string, 
  startDate: Date | null = null, 
  endDate: Date | null = null, 
  status: ProjectStatus = ProjectStatus.NOT_STARTED
): Project => ({
  id: uuid.v4().toString(),
  name,
  description,
  startDate,
  endDate,
  status,
  createdAt: new Date()
}); 